### PR TITLE
HUD Report Filters - Limit Projects by CoC Code

### PR DESCRIPTION
**Pull request by gigxz**
Update the CoC selector to act as a limit on `effective_project_ids`.

Screen Shot 2022-05-04 at 11 25 15 AM

## Affects the following reports

- Homeless Summary Report
- PATH report (also updated to use common view with project-filter stimulus component)
- APR
- CAPER
- CE APR
- HUD Data Quality Report
- HIC
- ~~LSA~~ 
- PIT
- SPM

*Ignore the branch name, accidentally pushed on wrong branch*

### CODE REVIEW

**Review by awisema — APPROVED** [review]

File: app/models/filters/hud_filter_base.rb [hunk 1]
```diff
@@ -30,6 +30,14 @@ def effective_project_ids
           pluck(p_t[:id])
       end
 
+      if coc_codes.present?
+        @effective_project_ids = GrdaWarehouse::Hud::ProjectCoc.in_cocs(coc_codes: coc_codes).
+          where(ProjectID: @effective_project_ids).
+          distinct.
+          pluck(:ProjectID).
```
Comment (eanders) [inline app/models/filters/hud_filter_base.rb]: ProjectID isn't correct, this needs to be `pluck(p_t[:id])`.  ProjectID is a string and is the internal foreign key for the particular data source.

File: app/models/grda_warehouse/hud/project_coc.rb [hunk 2]
```diff
@@ -38,6 +38,13 @@ class ProjectCoc < Base
       )
     end
 
+    scope :in_cocs, ->(coc_codes:) do
+      where(
+        pc_t[:CoCCode].in(coc_codes).and(pc_t[:hud_coc_code].eq(nil).or(pc_t[:hud_coc_code].eq(''))).
+        or(pc_t[:hud_coc_code].in(coc_codes)),
+      )
+    end
+
```
Comment (eanders) [inline app/models/grda_warehouse/hud/project_coc.rb]: Is there any functional difference between this and `scope :in_coc, ->(coc_code:)`?  They look identical, other than the use of plurals.

Comment (gigxz) [reply app/models/grda_warehouse/hud/project_coc.rb]: that's correct, I could modify the `in_coc` scope to accept either a single code or an array if that is preferable

Comment (eanders) [reply app/models/grda_warehouse/hud/project_coc.rb]: it already does.  `coc_code = Array(coc_code)` handles that.

File: app/models/filters/hud_filter_base.rb (hunk 1 above)
Comment (gigxz) [reply app/models/filters/hud_filter_base.rb]: ah this again. okay. is this right? spot checked with some specific projects
```
      @effective_project_ids = GrdaWarehouse::Hud::ProjectCoc.in_coc(coc_code: coc_codes).
          joins(:project).
          where(p_t[:id].in(@effective_project_ids)).
          distinct.
          pluck(p_t[:id])
```

Comment (eanders) [reply app/models/filters/hud_filter_base.rb]: That looks good.  I'd missed `where(ProjectID: @effective_project_ids).`  Thanks!

**Pull request merged**